lib/src/provider/network_vector_tile_provider.dart

```diff
@@ -6,16 +6,18 @@ import 'package:http/retry.dart';
 import '../../vector_map_tiles.dart';
 
 class NetworkVectorTileProvider extends VectorTileProvider {
```

Comment:
From the conversation it seems like changes to this class are unnecessary.

To address your concern around HTTP headers, the the factory function doesn't need to pass everything to the `VectorTileProvider` implementation - so that should help to keep things simple.